### DISCUSSION TITLE
modify `ls.available()` to include each snippets origin file path.

### DISCUSSION THREAD

**Discussion by molleweide** [L3MON4D3/LuaSnip#716] [Q&A]
## Proposed solution

After digging in the source a little bit I am thinking that the appropriate way to go about this would be 

1. Attach the `origin_file_path` to each snippet when loading files with `load_files(ft, files)` from `luasnip/loaders/from_lua.lua`. 
2. Update the `luasnip/init.lua` func `get_context()` so that it also grabs the path of each snip.
3. Modify the available function to mean "available in a broader sense".
    ```lua
    local function available(ft:<string|table>)        
    -- ft could be either a string or a table.
    -- 1. if ft == nil, then do the exact same as current `available()` and deprecate the available function.
    -- 2. if ft == string, retrieve snips for this specific file type.
    -- 3. if ft == table, then return all snippets for the containing filetypes.
    ```
4. optionally, also do this same procedure for `snipmate`, and `vscode` as well.
     
With these three additions it would be possible to extend the existing `telescope-luasnip` extension repo to become a CRUD manager. I also have some more ideas for the picker that would make life easier, eg. add snippet to new filetype and then create all of the boilerplate in your default luasnippets path for a new filetype.

If @L3MON4D3 if you agree with my proposed solution I will make a PR for this. This should NOT introduce any breaking changes.

**Comment by L3MON4D3** [comment]
Moved to #717
